main/manager.c

```diff
@@ -6873,6 +6900,203 @@ static int action_coreshowchannels(struct mansession *s, const struct message *m
 	return 0;
 }
 
+/*! \brief Destructor for channel map vector */
+static void channel_map_vector_destructor(void *obj)
+{
+	struct ast_vector_string *channel_map = obj;
+	AST_VECTOR_RESET(channel_map, ast_free);
+	AST_VECTOR_FREE(channel_map);
+}
+
+/*! \brief Helper function to add a channel name to the vector */
+static int coreshowchannelmap_add_to_map(struct ast_vector_string *v, const char *s)
+{
+	char *str;
+	int size;
+
+	/* If we've already added this channel, don't add it again */
+	if (AST_VECTOR_GET_CMP(v, s, !strcmp)) {
+                return -1;
+        }
+
+	size = strlen(s) + 1;
+	str = ast_malloc(size);
+	if (!str) {
+		ast_log(LOG_ERROR, "Unable to append channel to channel map\n");
+		return 1;
+	}
+
+	ast_copy_string(str, s, size);
+	if (AST_VECTOR_APPEND(v, str)) {
+		ast_log(LOG_ERROR, "Unable to append channel to channel map\n");
+		ast_free(str);
+		return 1;
+	}
+
+	return 0;
+}
+
+/*! \brief Recursive function to get all channels in a bridge. Follow local channels as well */
+static int coreshowchannelmap_add_connected_channels(struct ast_vector_string *channel_map,
+	struct ast_channel_snapshot *channel_snapshot, struct ast_bridge_snapshot *bridge_snapshot)
+{
+	int res = 0;
+	struct ao2_iterator iter;
+	char *current_channel_uid;
+
+	iter = ao2_iterator_init(bridge_snapshot->channels, 0);
+	while ((current_channel_uid = ao2_iterator_next(&iter))) {
+		struct ast_channel_snapshot *current_channel_snapshot;
+		int add_channel_res;
+
+		/* Don't add the original channel to the list - it's either already in there,
+		 * or it's the channel we want the map for */
+		if (!strcmp(current_channel_uid, channel_snapshot->base->uniqueid)) {
+			continue;
+		}
+
+		current_channel_snapshot = ast_channel_snapshot_get_latest(current_channel_uid);
+		if (!current_channel_snapshot) {
+			ast_log(LOG_ERROR, "Unable to get snapshot of channel\n");
+			res = 1;
+			break;
+		}
+
+		add_channel_res = coreshowchannelmap_add_to_map(channel_map, current_channel_snapshot->base->name);
+		if (add_channel_res > 0) {
+			ast_log(LOG_ERROR, "Unable to add channel to vector\n");
```

Comment:
coreshowchannelmap_add_to_map already logs an error if res > 0, so the additional error log here is redundant